saleor/graphql/payment/types.py

```diff
@@ -81,42 +81,82 @@ class Meta:
 
 class Payment(ModelObjectType):
     id = graphene.GlobalID(required=True)
-    gateway = graphene.String(required=True)
-    is_active = graphene.Boolean(required=True)
-    created = graphene.DateTime(required=True)
-    modified = graphene.DateTime(required=True)
-    token = graphene.String(required=True)
     checkout = graphene.Field("saleor.graphql.checkout.types.Checkout")
     order = graphene.Field("saleor.graphql.order.types.Order")
-    payment_method_type = graphene.String(required=True)
-    customer_ip_address = graphene.String()
-    charge_status = PaymentChargeStatusEnum(
-        description="Internal payment status.", required=True
-    )
     actions = graphene.List(
-        OrderAction,
+        graphene.NonNull(PaymentActionEnum),
         description=(
             "List of actions that can be performed in the current state of a payment."
         ),
         required=True,
     )
     total = graphene.Field(Money, description="Total amount of the payment.")
+    authorized_amount = graphene.Field(
+        Money, required=True, description="Total amount authorized for this payment."
+    )
+    refunded_amount = graphene.Field(
+        Money, required=True, description="Total amount refunded for this payment."
+    )
+    voided_amount = graphene.Field(
+        Money, required=True, description="Total amount voided for this payment."
+    )
     captured_amount = graphene.Field(
-        Money, description="Total amount captured for this payment."
+        Money, description="Total amount captured for this payment.", required=True
+    )
+
+    gateway = graphene.String(
+        required=True,
+        deprecation_reason=f"{DEPRECATED_IN_3X_FIELD} Use new checkout flow.",
+    )
+    is_active = graphene.Boolean(
+        required=True,
+        deprecation_reason=f"{DEPRECATED_IN_3X_FIELD} Use new checkout flow.",
+    )
+    created = graphene.DateTime(
+        required=True,
+    )
+    modified = graphene.DateTime(required=True)
+    token = graphene.String(
+        required=True,
+        deprecation_reason=f"{DEPRECATED_IN_3X_FIELD} Use new checkout flow.",
+    )
+    payment_method_type = graphene.String(
+        required=True,
+        deprecation_reason=f"{DEPRECATED_IN_3X_FIELD} Use new checkout flow.",
+    )
+    customer_ip_address = graphene.String(
+        deprecation_reason=f"{DEPRECATED_IN_3X_FIELD} Use new checkout flow."
+    )
+    charge_status = PaymentChargeStatusEnum(
+        description="Internal payment status.",
+        required=True,
+        deprecation_reason=f"{DEPRECATED_IN_3X_FIELD} Use new checkout flow.",
     )
     transactions = graphene.List(
-        Transaction, description="List of all transactions within this payment."
+        Transaction,
```

Comment:
```suggestion
        graphene.NonNull(Transaction),
```

Reply:
It is covered in separate PR:
https://github.com/saleor/saleor/pull/9391/files#diff-a5ec8861a37bb971cbfe0b5fff62d4c509e018f976e5735b6f2a37ee0f11bfc1R107